common/modules/farm/farm_token_merge/src/lib.rs

```diff
@@ -92,6 +96,10 @@ pub trait FarmTokenMergeModule:
 
             let attributes: FarmTokenAttributes<Self::Api> =
                 self.get_farm_token_attributes(&payment.token_identifier, payment.token_nonce);
+
+            let same_energy = first_token_attributes.energy.get_energy_amount() == attributes.energy.get_energy_amount();
+            let same_user = first_token_attributes.original_user == attributes.original_user;
+            require!(same_energy && same_user, ERROR_NOT_MERGEABLE);
```

Comment:
you have to check the EPOCH and TotalLKMEX as well. Add unit tests.

Reply:
Compared the whole Energy structure.